src/dunit/framework.d

```diff
@@ -293,6 +302,50 @@ body
                 foreach (testListener; testListeners)
                     testListener.exitTest(success);
 
+            if (test in testClass.enabledIf && !testClass.enabledIf.get(test, EnabledIf.init).condition)
```

Comment:
Why `.get` after the `in` check? Consider removing the `in` check:

    if (!testClass.enabledIf.get(test, EnabledIf(true)).condition)

Reply:
By following your advice and supporting condition evaluated at runtime, the additional `in` check is necessary because the `bool function() condition` field cannot be defaulted to `true` as in your example above.